website_user_login_redirect_custom/controllers/main.py

```diff
@@ -0,0 +1,75 @@
+# © 2025 Cetmix OU
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.web.controllers.main import Home
+
+
+class WebsiteRedirectCustom(Home):
+    """
+    Inherit from Home to override login behavior
+    This works even without auth_signup module
+    """
+
+    def _should_redirect_custom(self):
+        """Check if we should perform custom redirect"""
+        if not request.session.uid:
+            return False
+
+        # Do not redirect backend/internal users
+        if request.env.user.has_group("base.group_user"):
+            return False
+
+        enabled = (
+            request.env["ir.config_parameter"]
+            .sudo()
+            .get_param("website_user_login_redirect_custom.enabled", "0")
+        )
+        return enabled == "1"
+
+    def _get_custom_redirect_url(self):
+        """Get custom redirect URL or False if not set/valid"""
+        url = (
+            request.env["ir.config_parameter"]
+            .sudo()
+            .get_param("website_user_login_redirect_custom.url", "/")
+        )
+
+        if not url:
+            return False
+
+        url = url.strip()
+
+        # Validate URL format
+        if (
+            url.startswith("/")
+            and not url.startswith("//")
+            and "://" not in url
```

Comment:
You are doing URL validation in both controller and res.config settings. You can create a single function for that and then reuse it in all occasions.